prepreduce/prepreduce.py

```diff
@@ -0,0 +1,395 @@
+        user_inp = input(f"[🟡] target {file} already exists. Remove it? (y/n): ")
+        user_inp = user_inp.strip().lower()
+        if user_inp == "y":
+            break
+        elif user_inp == "n":
+            success("OK: quitting.")
+            exit(0)
+    info(f"Removing {file} which already exists")
+    file.unlink()
+
+
+def parse_cli_args() -> argparse.Namespace:
+    """
+    Parse out arguments from command line interface.
+    Use various validate_foo_from_cli functions to assist end-user
+    """
+    global DEBUG, VERSION
+
+    parser = argparse.ArgumentParser(
+        prog="prepreduce",
+        description=f"prepreduce v{VERSION}",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    parser.add_argument(
+        "target",
+        help="The source file you wish to target. Example: lib/string.o",
+        type=validate_target_from_cli,
+    )
+
+    parser.add_argument(
+        "-p",
+        "--path-to-linux",
+        type=lambda arg: validate_path_from_cli(arg, "-p/--path-to-linux"),
+        default="./",
+        help="Specify the Linux source tree directory",
+    )
+
+    parser.add_argument(
+        "build_command",
+        help="The specific build command preamble used sans the target. Example: `make -j$(nproc) LLVM=1 V=1`",
+        nargs="*",
+        default=["make", "-j8", "LLVM=1", "V=1"],
+    )
+
+    parser.add_argument(
+        "-f",
+        "--force-rm-existing-target",
+        help="If the target (foo.o) already exists, remove it -- without prompting",
+        action="store_true",
+        default=False,
+    )
+
+    parser.add_argument(
+        "-F",
+        "--force-rm-existing-script",
+        help="If test.sh already exists, remove it -- without prompting",
+        action="store_true",
+        default=False,
+    )
+
+    parser.add_argument(
+        "-d",
+        "--debug",
+        help="Enable prepreduce Debug logs",
+        action="store_true",
+        default=False,
+    )
+
+    parser.add_argument(
+        "-o",
+        "--output",
+        help="Output directory of prepreduce outputs (.i files and test scripts)",
+        default=os.getcwd(),
+        type=lambda arg: validate_path_from_cli(arg, "-o/--output", True, True),
+    )
+
+    args = parser.parse_args()
+    DEBUG = args.debug
+    if DEBUG:
+        ic.enable()
+
+    validate_build_cmd_from_cli(args.build_command)
+
+    return args
+
+
+def validate_build_cmd_from_cli(build_cmd: list[str]):
+    """
+    We need V=1 to find original clang invocation
+    """
+    try:
+        _ = build_cmd.index("V=1")  # raises ValueError if not found
+    except ValueError:
+        error("Need `V=1` in your make build command (suggested to use default)")
+
+
+def validate_target_from_cli(_target: str) -> Path:
+    """
+    Allow argparse to actually validate the target argument as well as convert
+    to PosixPath
+    """
+    try:
+        posix_path = Path(_target.strip())
+    except:
```

Comment:
Bare `except`s are a code smell IMO.

https://beta.ruff.rs/docs/rules/bare-except/

Even catching `Exception` or `BaseException` is frowned upon.

https://beta.ruff.rs/docs/rules/blind-except/

I have tried a bunch of different ways to generate an invalid path from a string and I can't do it. I would just remove this.